src/win/stream.c

```diff
@@ -48,6 +48,48 @@ int uv_listen(uv_stream_t* stream, int backlog, uv_connection_cb cb) {
 }
 
 
+int uv_reject(uv_stream_t* client) {
+  int err = 0;
+  uv_pipe_t* pipe_client;
+  uv_tcp_accept_t* req;
+
+  if (client == NULL) {
+    return uv_translate_sys_error(ERROR_INVALID_PARAMETER);
+  }
+
+  switch (client->type) {
+    case UV_NAMED_PIPE:
+      pipe_client = (uv_pipe_t*)client;
```

Comment:
Expose and use `close_pipe` rather than hand-roll the code here?

Reply:
Saúl, don't you think that is better to use something like in the UNIX side? Just call `uv_close` (not a win dev here)

https://github.com/libuv/libuv/blob/467859c2ba0beadbb536ebde919e3883c7b4bc11/src/win/handle.c#L84

That will handle the loop internals and will do a call to `close_pipe` too

https://github.com/libuv/libuv/blob/467859c2ba0beadbb536ebde919e3883c7b4bc11/src/win/pipe.c#L1099

Reply:
Yep, that looks good! Note that as I said elsewhere that is likely not what you want to do.

Reply:
Thanks for the review. From what I've seen, `uv_close()` closes all the connections (both queued and accepted). I thought we only needed to close the current request, not the accepted requests or queued requests. That's why I tried to implement `uv_reject()` using `closesocket()`. Am I missing something?

Reply:
If you look at the TCP an pipe implementations, they behave differently. One closes the queued accepted connection, the other one the server. The idea is call `uv_reject` on the _server_ to reject an incoming connection we couldn't handle. Thus, the server must remain up.

Reply:
I see what you mean. I've updated the `pipe` case to close the correct handle as seen in the attached patch. 
I'll try to fix other comments and provide a full patch to fix them all.
<details>
  <summary>Pipe patch</summary>

```diff
diff --git a/src/win/stream.c b/src/win/stream.c
index 2a98dcb2..0d2c1abb 100644
--- a/src/win/stream.c
+++ b/src/win/stream.c
@@ -48,40 +48,35 @@ int uv_listen(uv_stream_t* stream, int backlog, uv_connection_cb cb) {
 }
 
 
-int uv_reject(uv_stream_t* client) {
+int uv_reject(uv_stream_t* server) {
   int err = 0;
   uv_pipe_t* pipe_client;
-  uv_tcp_accept_t* req;
+  uv_tcp_accept_t* tcp_req;
+  uv_pipe_accept_t* pipe_req;
 
-  if (client == NULL) {
+  if (server == NULL) {
     return uv_translate_sys_error(ERROR_INVALID_PARAMETER);
   }
 
-  switch (client->type) {
+  switch (server->type) {
     case UV_NAMED_PIPE:
-      pipe_client = (uv_pipe_t*)client;
-      if (pipe_client->u.fd == -1) {
-        CloseHandle(pipe_client->handle);
-      } else {
-        _close(pipe_client->u.fd);
-      }
-
-      pipe_client->u.fd = -1;
-      pipe_client->handle = INVALID_HANDLE_VALUE;
+      pipe_req = ((uv_pipe_t*)server)->pipe.serv.pending_accepts;
+      CloseHandle(pipe_req->pipeHandle);
+      pipe_req->pipeHandle = INVALID_HANDLE_VALUE;
       break;
     case UV_TCP:
-      req = ((uv_tcp_t*)client)->tcp.serv.pending_accepts;
+      tcp_req = ((uv_tcp_t*)server)->tcp.serv.pending_accepts;
 
-      if (!req) {
+      if (!tcp_req) {
         err = WSAEWOULDBLOCK;
       }
 
-      if (req->accept_socket == INVALID_SOCKET) {
+      if (tcp_req->accept_socket == INVALID_SOCKET) {
         err = WSAENOTCONN;
       }
 
-      closesocket(req->accept_socket);
-      req->accept_socket = INVALID_SOCKET;
+      closesocket(tcp_req->accept_socket);
+      tcp_req->accept_socket = INVALID_SOCKET;
       break;
     default:
       err = ERROR_INVALID_PARAMETER;
```
</details>